src/components/assetlib_projects/asset_library_projects.gd

```diff
@@ -0,0 +1,352 @@
+const _ASSET_LISTING = preload("res://src/components/assetlib_projects/asset_listing_iteractable.tscn")
+
+var _current_page: int = 0:
+	get:
+		return _internal_current_page
+	set(value):
+		_internal_current_page = value
+		_fetch_assets(null, false)
+var _internal_current_page: int = 0
+var _current_assets: Dictionary: set = _display_assets
+var _last_text_edit: int = 0
+var _fetched_versions: bool = false:
+	set(value):
+		_fetched_versions = value
+		if not value:
+			_search_field.editable = false
+			_version_option.disabled = true
+			_sort_option.disabled = true
+			_category_option.disabled = true
+			_site_option.disabled = true
+		else:
+			_search_field.editable = true
+			_version_option.disabled = false
+			_sort_option.disabled = false
+			_category_option.disabled = false
+			_site_option.disabled = false
+
+@onready var _search_field: LineEdit = %SearchField
+@onready var _version_option: OptionButton = %VersionOption
+@onready var _sort_option: OptionButton = %SortOption
+@onready var _category_option: OptionButton = %CategoryOption
+@onready var _site_option: OptionButton = %SiteOption
+@onready var _support_options: MenuButton = %SupportOptions
+@onready var _asset_querier: HTTPRequest = $AssetQuerier
+@onready var _asset_list: HFlowContainer = %AssetList
+@onready var _navigation_buttons: HBoxContainer = %NavigationButtons
+@onready var _page_buttons: HBoxContainer = _navigation_buttons.get_node("PageButtons")
+@onready var _first_button: Button = _navigation_buttons.get_node("FirstButton")
+@onready var _previous_button: Button = _navigation_buttons.get_node("PreviousButton")
+@onready var _next_button: Button = _navigation_buttons.get_node("NextButton")
+@onready var _last_button: Button = _navigation_buttons.get_node("LastButton")
+@onready var _color_rect: ColorRect = %ColorRect
+@onready var _status_label: Label = %StatusLabel
+@onready var _refresh_button: Button = %RefreshButton
+@onready var _overlay_contents: CenterContainer = %OverlayContents
+
+
+func _ready():
+	$MarginContainer/ScrollContainer.add_theme_stylebox_override("panel",
+			get_theme_stylebox("search_panel", "ProjectManager"))
+	
+	_fetched_versions = false
+	
+	_support_options.get_popup().id_pressed.connect(_fetch_assets)
+	
+	# Populate PageButtons
+	for i in _MAX_PAGE_BUTTONS:
+		var button = Button.new()
+		button.name = "PageButton" + str(i)
+		_page_buttons.add_child(button)
+		button.pressed.connect(func():
+				_on_page_button_pressed(button)
+		)
+	
+	_first_button.pressed.connect(func():
+		_current_page = 0
+	)
+	_previous_button.pressed.connect(func():
+		_current_page = max(0, _current_page - 1)
+	)
+	_next_button.pressed.connect(func():
+		if _current_assets:
+			_current_page = min(_current_assets.pages - 1, _current_page + 1)
+		else:
+			_current_page = 0
+	)
+	_last_button.pressed.connect(func():
+		_current_page = _current_assets.pages - 1 if _current_assets else 0
+	)
+	
+	await _setup_version_button()
+	
+	_fetch_assets()
```

Comment:
_fetch_assets() has to be lazy. please take a look how this behaviour is implemented in RemoteEditors, the main idea is do not fetch something in a background, only if it is requested by the user (i.e. user changed a tab etc.)